src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -5069,7 +5069,8 @@ LogicalResult ONNXScanOp::inferShapes(
     // input to Loop operation, but we need to eliminate the possibility of
     // early termination to be sure.
     updateType(std::get<1>(vScanOutputValToTy), squeezedShape,
-        rankedScanTy.getElementType());
+        rankedScanTy.getElementType(), /*encoding=*/nullptr,
+        /*refineShape=*/false);
```

Comment:
should the updateType() call below in line 5113 also be called with refineShape=false?

Reply:
Good catch. I'll create a PR to update this. Thanks!